it/src/test/java/com/google/cloud/teleport/it/pubsub/DefaultPubsubResourceManagerTest.java

```diff
@@ -0,0 +1,236 @@
+    assertThat(actualTopicName)
+        .isEqualTo(TopicName.of(PROJECT_ID, String.format("%s%s", TEST_ID, "topic-name")));
+  }
+
+  @Test
+  public void testPublishMessage() {
+    // Arrange
+    when(publisher.publish(any(PubsubMessage.class)))
+        .thenReturn(ApiFutures.immediateFuture(VALID_MESSAGE_ID));
+    Map<String, String> attributes = ImmutableMap.of("key1", "value1");
+    ByteString data = ByteString.copyFromUtf8("valid message");
+
+    // Act
+    String publishMessage = testManager.publish(TOPIC_REFERENCE, attributes, data);
+
+    // Asert
+    assertThat(publishMessage).isEqualTo(VALID_MESSAGE_ID);
+    verify(publisher).publish(pubsubMessageCaptor.capture());
+    PubsubMessage actualMessage = pubsubMessageCaptor.getValue();
+    assertThat(actualMessage.getAttributesMap()).isEqualTo(attributes);
+    assertThat(actualMessage.getData()).isEqualTo(data);
+  }
+
+  @Test
+  public void testCleanupTopics() {
+    // Arrange
+    TopicName topicName1 = testManager.getTopicName("topic1");
+    TopicName topicName2 = testManager.getTopicName("topic2");
+    when(topic.getName()).thenReturn(topicName1.toString(), topicName2.toString());
+    when(topicAdminClient.createTopic(any(TopicName.class))).thenReturn(topic);
+
+    // Act
+    testManager.createTopic("topic1");
+    testManager.createTopic("topic2");
+    testManager.cleanupAll();
+
+    // Assert
+    verify(topicAdminClient, times(2)).deleteTopic(topicNameCaptor.capture());
+    assertThat(topicNameCaptor.getAllValues()).hasSize(2);
+    assertThat(topicNameCaptor.getAllValues())
+        .containsExactlyElementsIn(Arrays.asList(topicName1, topicName2));
+  }
+
+  @Test
+  public void testCleanupSubscriptions() {
+    // Arrange
+    SubscriptionName subscriptionName1 = testManager.getSubscriptionName("topic1-sub0");
+    SubscriptionName subscriptionName2 = testManager.getSubscriptionName("topic1-sub1");
+    SubscriptionName subscriptionName3 = testManager.getSubscriptionName("topic2-sub0");
+    when(subscription.getName())
+        .thenReturn(
+            subscriptionName1.toString(),
+            subscriptionName2.toString(),
+            subscriptionName3.toString());
+    when(subscriptionAdminClient.createSubscription(
+            any(SubscriptionName.class), any(TopicName.class), any(PushConfig.class), anyInt()))
+        .thenReturn(subscription);
+
+    // Act
+    testManager.createSubscription("topic1", "topic1-sub0");
+    testManager.createSubscription("topic1", "topic1-sub1");
+    testManager.createSubscription("topic2", "topic1-sub0");
+    testManager.cleanupAll();
+
+    // Assert
+    verify(subscriptionAdminClient, times(3)).deleteSubscription(subscriptionNameCaptor.capture());
+    assertThat(subscriptionNameCaptor.getAllValues()).hasSize(3);
+    assertThat(subscriptionNameCaptor.getAllValues())
+        .containsExactlyElementsIn(
+            Arrays.asList(subscriptionName1, subscriptionName2, subscriptionName3));
```

Comment:
nit: `containsExactly` is a little more readable, since you can just pass in the subscriptions without explicitly creating a collection: https://truth.dev/api/0.41/com/google/common/truth/IterableSubject.html#containsExactly-java.lang.Object...-